service/controller/clusterapi/v22/resource/clusterstatus/create.go

```diff
@@ -116,12 +116,14 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 	return nil
 }
 
-func (r *Resource) computeClusterConditions(ctx context.Context, cluster cmav1alpha1.Cluster, clusterStatus g8sv1alpha.CommonClusterStatus, nodes []corev1.Node, machineDeployments []cmav1alpha1.MachineDeployment) g8sv1alpha.CommonClusterStatus {
+func (r *Resource) computeClusterConditions(ctx context.Context, cc *controllercontext.Context, cluster cmav1alpha1.Cluster, clusterStatus g8sv1alpha.CommonClusterStatus, nodes []corev1.Node, machineDeployments []cmav1alpha1.MachineDeployment) g8sv1alpha.CommonClusterStatus {
```

Comment:
You should be able to extract controllercontext from context easily. I wouldn't pass it as argument.

Reply:
extracting controllercontext (again) can throw error, then have to change signature of the function to throw error, and handle it from call site - seemed like lots of noise for no obvious value, but will trust your guidance

Reply:
I see. I didn't realize it doesn't return error 👍